editoast/src/models/paced_train.rs

```diff
@@ -71,6 +71,19 @@ impl PacedTrain {
             options: self.options,
         }
     }
+
+    pub fn into_iter_occurrences(self) -> impl Iterator<Item = TrainSchedule> {
```

Comment:
```suggestion
    pub fn iter_occurrences(&self) -> impl Iterator<Item = TrainSchedule> {
```

you're still cloning anyway (I think the train_simulation_batch stack can be changed to take `&[&TrainSchedule]` and we'd avoid even more cloning, but let's keep this pR simple)

Reply:
I didn't see the comment about this, I'll close this one to keep the discussion there : https://github.com/OpenRailAssociation/osrd/pull/11355#discussion_r2053743491